optaplanner-persistence/optaplanner-persistence-jsonb/src/test/java/org/optaplanner/persistence/jsonb/api/score/AbstractScoreJsonbAdapterTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.persistence.jsonb.api.score;
+
+import java.io.Serializable;
+
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+import javax.json.bind.JsonbException;
+
+import org.optaplanner.core.api.score.Score;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class AbstractScoreJsonbAdapterTest {
+
+    // ************************************************************************
+    // Helper methods
+    // ************************************************************************
+
+    protected <S extends Score, W extends TestScoreWrapper<S>> void assertSerializeAndDeserialize(S expectedScore, W input) {
+        String jsonString;
+        W output;
+        try {
+            Jsonb jsonb = JsonbBuilder.create();
```

Comment:
In light of your recent changes, I was wondering... could we get rid of the `Test...ScoreWrapper`s, if we use the new `OptaplannerJsonbConfig` here, @cuijulian ?

Unless there's some technical hurdle I'm overlooking, this could remove a lot of code from this PR.

Reply:
1) Any such changes must also be applied to the jacksons tests (and potentially jaxb and xstream), for consistency.

2) Not sure about reusing the OptaPlannerJsonbConfig here: on one hand the philosophy of HardSoftScoreJsonbAdapterTest etc is to test the adaptor (= binding) in isolation, as a unit test. On the other hand, we want everyone to use the OptaPlannerJsonbConfig. So I am fine either way, as long is 1) is adhered to.

Reply:
This is currently consistent with the Jackson tests, any refactors will have to be done in a separate PR on all existing modules like Jackson, Jaxb, etc. As Geoffrey mentioned in his comment below, these individual Score Adapter tests are not quite a duplicate of the `OptaPlannerJsonbConfig` test. The latter tests that the Config object (which contains all the Score Adapters) actually works when used on a Wrapper with multiple Score types.